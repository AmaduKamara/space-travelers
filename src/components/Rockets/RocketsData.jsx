import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getRockets } from '../../redux/rockets/rockets';
import Rocket from './Rocket';

function RocketsData() {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rocketsReducer.rockets);

  useEffect(() => {
    dispatch(getRockets());
  }, []);

  return (
    <div className="my-5 px-5">
      {rockets.map((rocket) => (
        <Rocket
          rocket_name={rocket.rocket_name}
          rocket_type={rocket.rocket_type}
          rocket_image={rocket.flickr_images[0]}
          description={rocket.description}
          key={rocket.id}
        />
      ))}
    </div>
  );
}

export default RocketsData;
