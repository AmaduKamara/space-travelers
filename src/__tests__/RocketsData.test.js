import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import RocketsData from '../components/Rockets/RocketsData';
import store from '../redux/configStore';

const renderer = require('react-test-renderer');

it('should render the parent component', () => {
  const id = 1;
  const rocketName = 'Test Rocket';
  const rocketImage = 'img';
  const description = 'Test';
  const reserved = false;
  const handleToggleClick = () => true;
  const tree = renderer.create(
    <Provider store={store}>
      <RocketsData>
        id=
        {id}
        rocket_name=
        {rocketName}
        rocket_image=
        {rocketImage}
        description=
        {description}
        reserved=
        {reserved}
        handleToggleClick=
        {handleToggleClick}
      </RocketsData>
      {' '}

    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
