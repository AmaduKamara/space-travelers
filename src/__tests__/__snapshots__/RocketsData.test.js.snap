// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the parent component 1`] = `
Array [
  <div
    className="my-5 px-5 mb-16"
  />,
  " ",
]
`;
